app/assets/javascripts/beak/rewriter.coffee

```diff
@@ -0,0 +1,24 @@
+# So far I haven't had a reason to turn this into a real class or to add the properties as real code somewhere.  But I
+# still wanted the basic shape of the Rewriter documented somewhere.  Note that due to changes in how NetTango Web
+# handles recompiles on block changes, some of these are no longer used by that app.  -Jeremy B January 2023
```

Comment:
Since every single member of this non-class is optional... how does one tell if they have a `Rewriter`?  It doesn't look like it can be done either nominally or structurally.  It would be nice if there were *some* way (be it by making a real class, or adding some `type` member to the class to identify its type, or even an `isRewriter` property that serves as a witness to a value's `Rewriter`-iness).

If the argument is that we shouldn't care whether or not something is a `Rewriter`, then:

  1. 😒
  2. This file kind of begs the question of why it exists

Reply:
Sure, it can be an empty class for overriding for clarity.